### PR TITLE
MM-20873 Realm Migration for Channel Peek

### PR DESCRIPTION
**Pull request by imisshtml**
#### Summary
Realm migration for channel_peek screen.
Added markChannelViewedAndRead to realm/actions/channel.
Used the actions and selectors from the post realm files to support migration.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-20873

#### Device Information
This PR was tested on: 
iPhone X 13.2 Simulator

### CODE REVIEW

File: app/realm/utils/post.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ export function postDataToRealm(post, owner) {
 
     if (post.metadata?.reactions) {
         post.metadata.reactions.forEach((r) => {
-            const id = `${post.id}-${owner.id}-${r.emoji_name}`;
+            const id = `${post.id}-${owner?.id}-${r.emoji_name}`;
```
Comment (enahum) [inline app/realm/utils/post.js]: Why this change? This will create a string with `undefined`, definitely looks wrong

File: app/realm/actions/channel.js [hunk 2]
```diff
@@ -638,6 +639,12 @@ export function markChannelAsFavorite(channeId, favorite = true) {
     };
 }
 
+export function markChannelsViewedAndRead(channelId, previousChannelId, markOnServer = true) {
+    return () => {
+        reduxStore.dispatch(markChannelViewedAndRead(channelId, previousChannelId, markOnServer));
```
Comment (enahum) [inline app/realm/actions/channel.js]: Why is this needed? Aren’t we migrating?

File: app/screens/channel_peek/index.js [hunk 3]
```diff
@@ -1,37 +1,60 @@
...
+    const themePreferences = realm.objects('Preference').filtered(`category="${Preferences.CATEGORY_THEME}"`);
 
-    return {
-        channelId,
-        currentUserId: getCurrentUserId(state),
-        postIds: getPostIdsInChannel(state, channelId),
-        lastViewedAt: myMember && myMember.last_viewed_at,
-        theme: getTheme(state),
-    };
+    const currentUser = realm.objectForPrimaryKey('User', general?.currentUserId || '') || General.REALM_EMPTY_OBJECT;
```
Comment (enahum) [inline app/screens/channel_peek/index.js]: No need for the or in the query and at this point you already have the currentUserId, so no need to access it from general

File: app/screens/channel_peek/index.js [hunk 4]
```diff
@@ -1,37 +1,60 @@
...
+    const member = realm.objectForPrimaryKey('ChannelMember', `${channelId}-${currentUserId}`) || {addListener: () => null, removeListener: () => null};
```
Comment (enahum) [inline app/screens/channel_peek/index.js]: The listener should be replaced by the realm empty object constant

File: app/realm/utils/post.js (hunk 1 above)
Comment (imisshtml) [reply app/realm/utils/post.js]: The app would randomly crash due to owner being undefined at this point. It would happen randomly.

File: app/realm/actions/channel.js (hunk 2 above)
Comment (imisshtml) [reply app/realm/actions/channel.js]: Unless I read this wrong, this is updating the server side itself. Should I instead create an Action in the channel (realm/action/channel) that dispatches to client4 to update?

File: app/screens/channel_peek/index.js (hunk 3 above)
Comment (imisshtml) [reply app/screens/channel_peek/index.js]: Yes, missed that

File: app/screens/channel_peek/index.js (hunk 4 above)
Comment (imisshtml) [reply app/screens/channel_peek/index.js]: Good call, I pulled this from another setup so will update in both areas.

**Comment by jasonblais** [comment]
cc @migbot for next steps

**Comment by chuttam** [comment]
`ChannelPeek` was recently removed in 027c46a2 , so this PR is no longer necessary.
